File: Leetcode/Biweekly Contest/Biweekly Contest 132/3174.Clear Digits.cpp
class Solution
{
public:
    string clearDigits(string s)
    {
        int n = s.length();
        string res = "";
        int cnt = 0;
        for (int i = n - 1; i >= 0; i--)
        {
            if (isdigit(s[i]))
            {
                cnt++;
            }
            else
            {
                if (cnt == 0)
                    res += s[i];
                else
                {
                    cnt--;
                }
            }
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: Leetcode/Easy/3174. Clear Digits/3174. Clear Digits.py
class Solution:
    def clearDigits(self, s: str) -> str:
        """
        doing operation repeatedly 
        -> whenever see digit, if its adjacent is not digit -> remove
        -> use stack
        """
        
        stack = []

        for c in s:
            if c.isdigit():
                if stack: stack.pop()
            else:
                stack.append(c)

        return "".join(stack)